lib/rex/ui/text/input/buffer.rb

```diff
@@ -18,6 +18,10 @@ class BufferSock
     def write(buf, opts={})
       syswrite(buf)
     end
+
+    def monitor_rsock(*args, **kwargs)
+      dlog('monitor_rsock: Skipping - functionality not required')
+    end
```

Comment:
I believe this is the correct fix; Here was my code spelunking process:

HD Added this BufferSock class 16 years ago, and it depends on `StreamAbstraction`
https://github.com/rapid7/metasploit-framework/blame/5bf973871c3beca4788ec655b8ecd5e64af79e55/lib/rex/ui/text/input/buffer.rb#L16

The StreamAbstraction of 16 years ago, didn't have this `monitor_rsock` method call present - which runs in a loop automatically reading from rsock and writing the result:
https://github.com/rapid7/rex-core/commit/2970b21d3fb4dd30d529be6f6808dd1f170a4430#diff-78a17e4da58de200bc8f7afd1a44150d752a6a8756effe6478f0b589ece15f64

Then recently there was an implementation update to `monitor_rsock` which automatically closes things when the monitor loop ends:

https://github.com/rapid7/rex-core/commit/caa60274ec93a1597a4fe0d1d032fd8575593a9d#diff-bf61a62a028b4adbe2af6a11f60e8b3aa226ce3f1101be48330cf16f0f4ac98aR211

The loop was exiting as an `Errno::EAGAIN` exception was being raised on the read call [here](https://github.com/rapid7/rex-core/blob/caa60274ec93a1597a4fe0d1d032fd8575593a9d/lib/rex/io/socket_abstraction.rb#L157), I believe this is because there was competition for reading from the monitor loop and the normal msfconsole webconsole pipe reader.

TL:DR The default `monitor_rsock` implementation shouldn't be running in this scenario, and it now forcibly kills the webconsole pipe when an error is raised